### PR TITLE
Refactor rnp_cfg_t usage within cli_rnp_t

### LINKED ISSUE #1075: kr-def-key parameter ignored
## Description

I would expect to sign (or do some other operation) with default key if the key is not specified for the operation. However, `kr-def-key` configuration value never gets to `cli_rnp_t.defkey` field.

## Steps to Reproduce
???

## Expected Behavior
Value from `CFG_KR_DEF_KEY` should go to `cli_rnp_t.defkey`

## Actual Behavior
Value from `CFG_KR_DEF_KEY` is never used.
Always trying to use the first key in the keystore.

### PR DESCRIPTION
**Pull request by ni4**
This PR improves usage of rnp_cfg_t for cli_rnp_t structure, which would make implementation of future CLI features easier.
Also fixes #1075

### CODE REVIEW

**Comment by ronaldtse** [comment]
Ping @rrrooommmaaa @dewyatt for a review. Thanks!

**Comment by dewyatt** [comment]
I see that `rnp_cfg_t` is moved inside of `cli_rnp_t`. Is the goal just to get rid of an extra parameter to some functions, or to keep everything CLI-related in `cli_rnp_t`, or what?
Just looking for some overall rationale since this is a decent sized set of changes.

**Comment by ni4** [comment]
> Is the goal just to get rid of an extra parameter to some functions, or to keep everything CLI-related in cli_rnp_t, or what?
@dewyatt actually both, and additionally to isolate CLI RNP variables/fields from 'raw' parameters passed from the command line.

File: src/tests/key-unlock.cpp [hunk 1]
```diff
@@ -152,8 +149,7 @@ TEST_F(rnp_tests, test_key_unlock_pgp)
     verf.seekg(versize - 3, std::ios::beg);
     verf.write("0x0C", 1);
```
[the remark this answers is not in the document]
Comment (ni4) [reply src/tests/key-unlock.cpp]: @rrrooommmaaa Good catch. Let's fix it here instead of creating separate PR.

**Pull request merged**